plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2425,6 +2430,9 @@ public ConnectorTableHandle beginDelete(ConnectorSession session, ConnectorTable
         if (retryMode != NO_RETRIES) {
             throw new TrinoException(NOT_SUPPORTED, "Deleting from Hive tables is not supported with query retries enabled");
         }
+        if (isTransactionalTable(table.getParameters()) && !autoCommit) {
+            throw new TrinoException(NOT_SUPPORTED, "Deleting from Hive transactional tables is not supported in explicit transactions (use autocommit mode)");
```

Comment:
remove `isTransactionalTable` -- `beginDelete` won't be called for a non-acid table.

Reply:
I just tried the scenario of deleting from a non-transactional table that is being partitioned and this worked.

```
CREATE TABLE test2 (a_string varchar, a_integer int)
WITH (format = 'ORC', transactional = false, partitioned_by = ARRAY['a_integer']);

insert into test2 values ('hello', 1);
insert into test2 values ('world', 2);
insert into test2 values ('Trino', 1);

delete from test2 where a_integer =1 ;
```

I tend to think that the check makes sense to be kept.

Reply:
Are you saying that removing `isTransactionalTable` check here would break this scneario?

Reply:
The example above would call `applyDelete`, righ?

Reply:
@losipiuk yes

Reply:
```
trino:default> start transaction;
START TRANSACTION
trino:default> delete from test2 where a_integer =1 ;
DELETE

Query 20220128_112521_00030_bfch6, FINISHED, 1 node
http://localhost:8080/ui/query.html?20220128_112521_00030_bfch6
Splits: 1 total, 1 done (100.00%)
CPU Time: 0.0s total,     0 rows/s,     0B/s, 100% active
Per Node: 0.0 parallelism,     0 rows/s,     0B/s
Parallelism: 0.0
Peak Memory: 116B
0.43 [0 rows, 0B] [0 rows/s, 0B/s]

trino:default> select * from test2;
 a_string | a_integer 
----------+-----------
 world    |         2 
(1 row)

Query 20220128_112531_00031_bfch6, FINISHED, 1 node
http://localhost:8080/ui/query.html?20220128_112531_00031_bfch6
Splits: 9 total, 9 done (100.00%)
CPU Time: 0.0s total,   166 rows/s,   49KB/s, 46% active
Per Node: 0.0 parallelism,     3 rows/s,   965B/s
Parallelism: 0.0
Peak Memory: 150B
0.31 [1 rows, 301B] [3 rows/s, 965B/s]

trino:default> rollback;
ROLLBACK
trino:default> select * from test2;
 a_string | a_integer 
----------+-----------
 hello    |         1 
 Trino    |         1 
 world    |         2 
(3 rows)

Query 20220128_112538_00033_bfch6, FINISHED, 1 node
http://localhost:8080/ui/query.html?20220128_112538_00033_bfch6
Splits: 11 total, 11 done (100.00%)
CPU Time: 0.0s total,   187 rows/s, 54.9KB/s, 42% active
Per Node: 0.0 parallelism,     6 rows/s, 1.92KB/s
Parallelism: 0.0
Peak Memory: 150B
0.46 [3 rows, 900B] [6 rows/s, 1.92KB/s]
```

I've tried this functionality on `master` where the check is not present.

Reply:
```
trino:default> CREATE TABLE txtest (a_string varchar) WITH (format = 'ORC', transactional = true);
CREATE TABLE
trino:default> insert into txtest values ('a'), ('b'), ('c');
INSERT: 3 rows

Query 20220128_113930_00016_cj4we, FINISHED, 1 node
http://localhost:8080/ui/query.html?20220128_113930_00016_cj4we
Splits: 19 total, 19 done (100.00%)
CPU Time: 0.0s total,     0 rows/s,     0B/s, 7% active
Per Node: 0.0 parallelism,     0 rows/s,     0B/s
Parallelism: 0.0
Peak Memory: 416B
1.08 [0 rows, 0B] [0 rows/s, 0B/s]

trino:default> start transaction;
START TRANSACTION
trino:default> delete from txtest where a_string = 'a';
DELETE: 1 row

Query 20220128_113950_00018_cj4we, FINISHED, 1 node
http://localhost:8080/ui/query.html?20220128_113950_00018_cj4we
Splits: 10 total, 10 done (100.00%)
CPU Time: 0.1s total,    57 rows/s,   13KB/s, 50% active
Per Node: 0.1 parallelism,     6 rows/s, 1.36KB/s
Parallelism: 0.1
Peak Memory: 108B
0.50 [3 rows, 693B] [6 rows/s, 1.36KB/s]

trino:default> delete from txtest where a_string = 'b';
Query 20220128_113954_00019_cj4we failed: Tried to access metastore after transaction has been committed/aborted
java.lang.IllegalStateException: Tried to access metastore after transaction has been committed/aborted
	at io.trino.plugin.hive.metastore.SemiTransactionalHiveMetastore.checkReadable(SemiTransactionalHiveMetastore.java:2441)
	at io.trino.plugin.hive.metastore.SemiTransactionalHiveMetastore.getTable(SemiTransactionalHiveMetastore.java:217)
	at io.trino.plugin.hive.HiveMetadata.getView(HiveMetadata.java:2391)
```

Tested on `config-hdp3`

Reply:
i  fail to understand the response.
it would help me if you did remove the check and the CI showed the failure for me.

Reply:
@findinpath just dro the `isTransactional` part of check.
It is not needed. It is already checked above via `ensureTableSupportsDelete()`

Reply:
I would not be strongly against `verify(isTransactionalTable(table.getParameters())` but it feels overchatty.